### ISSUE TITLE
npm only build 6 weights and throw an error

### ISSUE THREAD

**Issue by MisaelVillaverde** [be5invis/Iosevka#847]
![image](https://user-images.githubusercontent.com/30275501/106325451-d83d5e80-6248-11eb-8aea-7fb604b0daaa.png)

I tried with docker too but it doesn't seem to work:
![image](https://user-images.githubusercontent.com/30275501/106325584-0a4ec080-6249-11eb-9fd4-078aa6bca862.png)

---

* Your font version: latest(5.0.0)
* Your font variant: Custom
* Your operating system (name and version): Ubuntu 20.04.1 (WSL2)
* Your application using Iosevka: 

If you have problem when building Iosevka, please provide these information:

* Your Node.js version:  v15.7.0
* Your _otfcc_ version: 0.10.4
* Your `params/parameters.toml`, paste if modified:
  ```
  Not modified
  ```
* Your `build-plans.toml`, paste if modified:
  ```
  Not modified
  ```
* Your `private-build-plans.toml`, if present:
  ```
  [buildPlans.iosevka-custom]
  family = "Iosevka Custom"
  spacing = "normal"
  serifs = "sans"
  
    [buildPlans.iosevka-custom.variants]
    inherits = "ss08"
  
      [buildPlans.iosevka-custom.variants.design]
      lig-ltgteq = "slanted"
  
    [buildPlans.iosevka-custom.ligations]
    inherits = "haskell"
  ```
* The detailed error log:
```
  0 verbose cli [
  0 verbose cli   '/home/misaelvillaverde/.nvm/versions/node/v15.7.0/bin/node',
  0 verbose cli   '/home/misaelvillaverde/.nvm/versions/node/v15.7.0/bin/npm',
  0 verbose cli   'run',
  0 verbose cli   'build',
  0 verbose cli   '--',
  0 verbose cli   'ttf::iosevka-custom'
  0 verbose cli ]
  1 info using npm@7.4.3
  2 info using node@v15.7.0
  3 timing config:load:defaults Completed in 1ms
  4 timing config:load:file:/home/misaelvillaverde/.nvm/versions/node/v15.7.0/lib/node_modules/npm/npmrc Completed in 0ms
  5 timing config:load:builtin Completed in 0ms
  6 timing config:load:cli Completed in 1ms
  7 timing config:load:env Completed in 1ms
  8 timing config:load:file:/mnt/d/fonts/Iosevka/Iosevka/.npmrc Completed in 4ms
  9 timing config:load:project Completed in 6ms
  10 timing config:load:file:/home/misaelvillaverde/.npmrc Completed in 0ms
  11 timing config:load:user Completed in 0ms
  12 timing config:load:file:/home/misaelvillaverde/.nvm/versions/node/v15.7.0/etc/npmrc Completed in 1ms
  13 timing config:load:global Completed in 1ms
  14 timing config:load:cafile Completed in 0ms
  15 timing config:load:validate Completed in 0ms
  16 timing config:load:setUserAgent Completed in 0ms
  17 timing config:load:setEnvs Completed in 1ms
  18 timing config:load Completed in 11ms
  19 verbose npm-session b627a441a200c3b2
  20 timing npm:load Completed in 23ms
  21 timing command:run-script Completed in 137648ms
  22 verbose stack Error: command failed
  22 verbose stack     at ChildProcess.<anonymous> (/home/misaelvillaverde/.nvm/versions/node/v15.7.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
  22 verbose stack     at ChildProcess.emit (node:events:379:20)
  22 verbose stack     at maybeClose (node:internal/child_process:1069:16)
  22 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:300:5)
  23 verbose pkgid iosevka@5.0.0-beta.2
  24 verbose cwd /mnt/d/fonts/Iosevka/Iosevka
  25 verbose Linux 5.4.72-microsoft-standard-WSL2
  26 verbose argv "/home/misaelvillaverde/.nvm/versions/node/v15.7.0/bin/node" "/home/misaelvillaverde/.nvm/versions/node/v15.7.0/bin/npm" "run" "build" "--" "ttf::iosevka-custom"
  27 verbose node v15.7.0
  28 verbose npm  v7.4.3
  29 error code 1
  30 error path /mnt/d/fonts/Iosevka/Iosevka
  31 error command failed
  32 error command sh -c node utility/ensure-verda-exists && verda -f verdafile.js "ttf::iosevka-custom"
  33 verbose exit 1
```

**Comment by be5invis** [comment]
Could you try run this:

```bash
npm run build -- --verbosity=7 ttf::iosevka-custom
```

**Comment by MisaelVillaverde** [comment]
> Could you try run this:
> 
> ```shell
> npm run build -- --verbosity=7 ttf::iosevka-custom
> ```

thanks, this time the output was:
![image](https://user-images.githubusercontent.com/30275501/106340715-c4095980-6268-11eb-965f-2c9ad4a05e21.png)

**Comment by be5invis** [comment]
Weird it doesn't produce any message.
I cannot reproduce this locally (with latest `npm` dependencies and the exact same `Node` version).

**Comment by MisaelVillaverde** [comment]
:( ok, I managed to run it in docker using `sudo docker run -e FONT_VERSION=5.0.0-beta.2 -it -v $(pwd):/build avivace/iosevka-build ttf::iosevka-custom --verbosity=7` and the output was: 
![image](https://user-images.githubusercontent.com/30275501/106343741-e6a16f80-6274-11eb-8155-b6a9b65c6a0b.png)

**Comment by pv4** [comment]
@MisaelVillaverde

> 25 verbose Linux 5.4.72-microsoft-standard-WSL2

According to this the build is done in WSL, not native Linux. I haven't used Windows for building iosevka but maybe native Windows build (instead of WSL) would help?

Other possible reason is not enough RAM (try monitoring it while building).

Also, if supported by WSL you could run the build under strace (`strace -fo build.strace npm run build ttf::iosevka-custom`) and look through the log to find the lowlevel root of the issue.

@be5invis , is it possible to build Iosevka on Windows natively, not in WSL?

**Comment by be5invis** [comment]
@pv4 The building worked fine on Windows.

**Comment by be5invis** [comment]
One thing may worth noticing is that building Iosevka may need a lot of memory. Maybe OOM would trigger such failure.

**Comment by MisaelVillaverde** [comment]
> One thing may worth noticing is that building Iosevka may need a lot of memory. Maybe OOM would trigger such failure.

Oh ok, that's weird because I've 16gb of RAM, maybe it's a problem with wsl, I will try building in windows. I tried yesterday with a linux partition I have, it created 12 ttfs and then the terminal start consuming all RAM and crashed the system, I had to hard boot, I tried like 3 times with different terminals and same happens(Maybe it's a problem with my hardware too). I will try with windows. Thanks.

**Comment by alexeyten** [comment]
> I tried yesterday with a linux partition I have, it created 12 ttfs and then the terminal start consuming all RAM and crashed the system,

Don’t your terminal uses the same font as one you’re trying to build? Mine crashes in that case I and I have to temporarily switch terminal font to something else.

**Comment by MisaelVillaverde** [comment]
> Don’t your terminal uses the same font as one you’re trying to build? Mine crashes in that case I and I have to temporarily switch terminal font to something else.

In wsl I use Iosevka default variant, but in the linux partition(pop os) I used SF Mono, so it doesn't have to do I guess, I will change the wsl one to another, that's weird tho.

**Comment by MisaelVillaverde** [comment]
I finally build it in powershell, I didn't even try that in the first place, sorry for that I'm used to use wsl instead of cmd or powershell. I had to close everything thinking it would crash but it worked, thanks for the help.
![image](https://user-images.githubusercontent.com/30275501/106363897-4e010300-62f9-11eb-8ffe-3d3f06da720f.png)

**Comment by lorenzogatti** [comment]
Does the build tool have some option similar to -j in make, to control the number of jobs that run in parallel instead of saturating all processor cores? Reducing the number of jobs would help with RAM usage in addition to sparing a portion of the CPU.

**Comment by be5invis** [comment]
@lorenzogatti There's `--jCmd`, and you can do things like

```bash
npm run build -- --jCmd=16 ttf::iosevka
```

**Comment by pv4** [comment]
@lorenzogatti On *nix it's also possible to use `nice` command to run another command (build process in our case) with lower priority. E.g.
```sh
nice npm run build -- ttf::iosevka
```

**Comment by lorenzogatti** [comment]
Thank you, I can confirm that -jCmd works perfectly on Windows 10 (with pnpm replacing npm).

**Comment by be5invis** [comment]
Resolve as fixed.
